pkg/resources/secret_with_basic_authentication.go

```diff
@@ -0,0 +1,163 @@
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the username value to store in the secret when setting the TYPE value to PASSWORD.",
+		},
+		"password": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the password value to store in the secret when setting the TYPE value to PASSWORD.",
+		},
+	}
+	return helpers.MergeMaps(secretCommonSchema, secretBasicAuthentication)
+}()
+
+func SecretWithBasicAuthentication() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateContextSecretWithBasicAuthentication,
+		ReadContext:   ReadContextSecretWithBasicAuthentication,
+		UpdateContext: UpdateContextSecretWithBasicAuthentication,
+		DeleteContext: DeleteContextSecretWithBasicAuthentication,
+
+		Schema: secretBasicAuthenticationSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Description: "Secret with Basic Authentication where Secrets Type attribute is set to PASSWORD.",
+	}
+}
+
+func CreateContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	commonCreate := handleSecretCreate(d)
+
+	id := sdk.NewSchemaObjectIdentifier(commonCreate.database, commonCreate.schema, commonCreate.name)
+
+	usernameString := d.Get("username").(string)
+	passwordString := d.Get("password").(string)
+
+	request := sdk.NewCreateWithBasicAuthenticationSecretRequest(id, usernameString, passwordString)
+	if v, ok := d.GetOk("comment"); ok {
+		request.WithComment(v.(string))
+	}
+
+	err := client.Secrets.CreateWithBasicAuthentication(ctx, request)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeResourceIdentifier(id))
+
+	return ReadContextSecretWithBasicAuthentication(ctx, d, meta)
+}
+
+func ReadContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	secret, err := client.Secrets.ShowByID(ctx, id)
+	if secret == nil || err != nil {
+		if errors.Is(err, sdk.ErrObjectNotFound) {
+			d.SetId("")
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Warning,
+					Summary:  "Failed to retrieve secret. Target object not found. Marking the resource as removed.",
+				},
+			}
+		}
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to retrieve secret.",
+				Detail:   fmt.Sprintf("Id: %s\nError: %s", d.Id(), err),
+			},
+		}
+	}
+	secretDescription, err := client.Secrets.Describe(ctx, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if err := handleSecretRead(d, id, secret, secretDescription); err != nil {
+		return diag.FromErr(err)
+	}
+	if err = d.Set("username", secretDescription.Username); err != nil {
```

Comment:
If the username is returned I think it may be not sensitive, it's also stored in the user so you can check it there or ask @sfc-gh-asawicki.

Reply:
`display_name` in user is not sensitive, i think it shouldn't be sensitive(?) since it is being returned as you've mentioned

Reply:
Display name is just the name that is displayed but login_name is the one used to login to snowsight and as you can see it is sensitive